### ISSUE TITLE
Make test_tif_loader publish quicker by taking advantage of transient local QOS

### ISSUE THREAD

**Issue by Ryanf55** [ethz-asl/grid_map_geo#58]
# Purpose

The test_tif_loader has a 5 second timer to publish the grid map. The way timers work in ROS is that they aren't executed when constructed; they must go through the first period of time.

The effect is that there is always a 5 second wait before the map shows up in RVIZ.

# Goal for this ticket

Make the map publish earlier, as soon as it's ready to be published.

# Hints

What if you call `timer_callback();` at the end of the `MapPublisher` constructor - seems like it publishes before RVIZ is up and ready to receive the message. 
What if you change the quality of service of the publisher to reliable, history length 1, transient local durability to support the following behavior:
```
To achieve a “latched” topic that is visible to late subscribers, both the publisher and subscriber must agree to use ‘Transient Local’.
```
Finally, if the QoS is changed on the publisher, rviz should be updated to do the same.

# Longer term

grid_map_geo should only publish the map when it changes, rather than on a timer, and take advantage of ROS QoS. Is there a reason we aren't using transient local QoS for the map?

# Reference

* https://docs.ros.org/en/humble/Concepts/Intermediate/About-Quality-of-Service-Settings.html#qos-profiles

**Comment by Jaeyoung-Lim** [comment]
@Ryanf55 In terrain-navigation, we use a latched topic, so it is only publsihed when subscribed.

For the map publisher node, I think since we are not updating the map in the node, it is actually irrelevant

**Comment by Ryanf55** [comment]
> @Ryanf55 In terrain-navigation, we use a latched topic, so it is only publsihed when subscribed.
> 
> For the map publisher node, I think since we are not updating the map in the node, it is actually irrelevant

Right. What about the map publisher being latched and just remove the timer on publish for consistency in behavior. Thisi will have behavior because as soon as RVIZ boots, it gets the map without the delay it has now.
